### ISSUE TITLE
REPL namespace incorrectly reported after executing a REPL command in another namespace

### ISSUE THREAD

**Issue by danielcompton** [cursive-ide/cursive#1624]
After running a REPL command that executes in a specific namespace, the namespace indicator in a remote repl shows the namespace that the command was executed in, not the one it is currently in. If you run a command in the REPL, then it will detect the correct namespace. E.g.:

<img width="498" alt="screenshot of intellij idea 4-01-17 4-42-11 pm" src="https://cloud.githubusercontent.com/assets/811954/21631132/c38d5692-d29c-11e6-9ace-4f42652e6a90.png">

<img width="530" alt="screenshot of intellij idea 4-01-17 4-44-35 pm" src="https://cloud.githubusercontent.com/assets/811954/21631158/159d82f4-d29d-11e6-8c87-b06a45159f70.png">
<img width="453" alt="screenshot of intellij idea 4-01-17 4-44-57 pm" src="https://cloud.githubusercontent.com/assets/811954/21631172/227d427a-d29d-11e6-80c3-87f345ff6fdf.png">

**Comment by danielcompton** [comment]
I don't think I got this bug report quite right. The namespace indicator for the REPL is correct, the problem is that if you run a REPL command in a different ns, the namespace itself isn't switched back until after you run a further command in that REPL. This leads to confusing behaviour when you're toggling between a REPL command, and trying to run a function in a particular namespace.

**Comment by danielcompton** [comment]
This is still an issue in the latest builds. What I think is happening is:

1. I'm in a namespace (`myapp.core`)
2. I run a REPL command which executes in the `user` namespace
3. After the command is finished, the namespace indicated stays in the `user` namespace. Any suggestions that Cursive gives in the REPL command window are all for the `user` namespace. However if you actually try to access any vars in the user namespace, you will get an error because the REPL is running in the `myapp.core` namespace, but Cursive hasn't caught up to the change.
4. After running any command, the REPL namespace in the Cursive REPL window updates to show the `myapp.core` namespace.

In summary, it looks like the issues is that Cursive isn't detecting the second namespace switch away from the `user` namespace when running a REPL command.

**Comment by cursive-ide** [comment]
I've just tested this, and it now works correctly. Since a lot has changed since this issue was filed, I'm going to assume this got fixed along the way. Let me know if this is not the case and this should be reopened.
